src/empty.rs

```diff
@@ -0,0 +1,51 @@
+use super::Body;
+use bytes::Buf;
+use http::HeaderMap;
+use std::{
+    convert::Infallible,
+    marker::PhantomData,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+/// A body that is always empty.
+#[derive(Debug, Clone, Copy)]
+pub struct Empty<D> {
+    _marker: PhantomData<D>,
+}
+
+impl<D> Default for Empty<D> {
+    fn default() -> Self {
+        Self {
+            _marker: PhantomData,
+        }
+    }
+}
+
+impl<D> Empty<D> {
+    /// Create a new `Empty`.
+    pub fn new() -> Self {
+        Self::default()
+    }
```

Comment:
Structs of this kind are usually created by a freestanding function rather than an associated function (cf. [`std::future::pending`][pending] and [`futures_util::stream::empty`][empty]).

[pending]: https://doc.rust-lang.org/1.49.0/std/future/fn.pending.html
[empty]: https://docs.rs/futures-util/0.3.12/futures_util/stream/fn.empty.html

Reply:
I'm open to that but its not my personal preference. I would like to hear what someone else thinks.